src/couch_mrview/src/couch_mrview_http.erl

```diff
@@ -513,8 +518,20 @@ parse_body_and_query(Req, Keys) ->
 
 parse_body_and_query(Req, {Props}, Keys) ->
     Args = #mrargs{keys = Keys, group = undefined, group_level = undefined},
-    BodyArgs = parse_params(Props, Keys, Args, [decoded]),
-    parse_params(chttpd:qs(Req), Keys, BodyArgs, [keep_group_level]).
+    BodyArgs0 = parse_params(Props, Keys, Args, [decoded]),
+    BodyArgs1 = treat_single_keys_as_key(Req, BodyArgs0),
+    parse_params(chttpd:qs(Req), Keys, BodyArgs1, [keep_group_level]).
+
+treat_single_keys_as_key(#httpd{path_parts = Path}, #mrargs{keys = Keys} = BodyArgs) ->
+    case lists:member(<<"_view">>, Path) of
```

Comment:
`_view` only has a special meaning in certain path parts, so this isn't quite right. I think we'll need a caluse that matches `_view` in the correct place and then another clause that doesn't look at path_parts and just returns `BodyArgs` (for the _all_docs case, right?).

Reply:
This way is much safer than just checking if `_view` exists in the path parts, thanks for the suggestion.